samples/wsdlclient4.php

```diff
@@ -20,15 +20,17 @@
 $proxyport = isset($_POST['proxyport']) ? $_POST['proxyport'] : '';
 $proxyusername = isset($_POST['proxyusername']) ? $_POST['proxyusername'] : '';
 $proxypassword = isset($_POST['proxypassword']) ? $_POST['proxypassword'] : '';
+$client = new soapclient('http://www.scottnichol.com/samples/round2_base_server.php?wsdl&debug=1', true,
+						$proxyhost, $proxyport, $proxyusername, $proxypassword);
 /*
  *	When no method has been specified, give the user a choice
  */
 if ($method == '') {
 	echo '<form name="MethodForm" method="POST">';
-	echo '<input type="hidden" name="proxyhost" value="' . $proxyhost .'">';
-	echo '<input type="hidden" name="proxyport" value="' . $proxyport .'">';
-	echo '<input type="hidden" name="proxyusername" value="' . $proxyusername .'">';
-	echo '<input type="hidden" name="proxypassword" value="' . $proxypassword .'">';
+	echo '<input type="hidden" name="proxyhost" value="' . $client->sanitize($proxyhost) .'">';
+	echo '<input type="hidden" name="proxyport" value="' . $client->sanitize($proxyport) .'">';
+	echo '<input type="hidden" name="proxyusername" value="' . $client->sanitize($proxyusername) .'">';
+	echo '<input type="hidden" name="proxypassword" value="' . $client->sanitize($proxypassword) .'">';
```

Comment:
This should probably just use `htmlspecialchars` directly, because we shouldn't be changing the value, just escaping it for the browser.